Shopify/react-native-skia
# #652 - iOS: Fixed podspec to support use_frameworks

When using the Firebase library we are required to add `use_frameworks! :linkage => :static` to the Pod file of the consuming app. This has until now caused some issues with building Skia on iOS (a workaround exists, but it shouldn't be necessary).

The reason for the problem is that RN Skia uses subspecs with header mappings (we are trying to preserve some of the file structure to make our external Skia includes build correctly) which isn't used when use_frameworks are set. This causes XCode to not find the include files.

This commit fixes this by removing the subspec and adding header search paths for the Skia pod. Only the pod file is changed, so this will not affect Android build.

**Testplan:**

1. Run `pod-install` in the example and verify that Xcode builds the example correctly.
2. Add `use_frameworks! :linkage => :static` to Example/iOS/Podfile (line 5-6) and disable Flipper (comment out line 42)
3. Run `pod-install` in the example and verify that Xcode builds the example correctly.

fixes #652 
fixes #1395
